From a73f30dcfe746079ab6e9b851ce7576dc1f4f50b Mon Sep 17 00:00:00 2001
From: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Date: Tue, 3 Sep 2013 12:10:42 -0700
Subject: Dalvik: Klocwork: Vectorization part 2

BZ: 135703

There was a missing nil pointer check for klocwork to be happy.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Iafd0b449f1f2ec4869d337981f07489357fa80be
Orig-MCG-Change-Id: I69054ccbd59ba2e379a30679dd32ddf7e398c16b
Signed-off-by: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Reviewed-on: http://android.intel.com:8080/131194
Reviewed-by: Banerji, Udayan <udayan.banerji@intel.com>
Reviewed-by: Katkov, Serguei I <serguei.i.katkov@intel.com>
Reviewed-by: Lupusoru, Razvan A <razvan.a.lupusoru@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Reviewed-by: Popov, Ivan G <ivan.g.popov@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/Vectorization.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/vm/compiler/Vectorization.cpp b/vm/compiler/Vectorization.cpp
index 0ee64ce..ee9f5b5 100644
--- a/vm/compiler/Vectorization.cpp
+++ b/vm/compiler/Vectorization.cpp
@@ -1857,6 +1857,13 @@ static bool vectorizeHelper (CompilationUnit *cUnit, LoopInformation *loopInform
     //We want the vectorized test
     BasicBlock *vectorizedTest = createVectorizedTest (cUnit, loopInformation, info);
 
+    //Be sure the basic blocks have been created
+    if (copyBasicBlock == 0 || copyPreHeader == 0 || copyExit == 0 || copyBWCC == 0 || vectorizedTest == 0 || mainTest == 0)
+    {
+        //We let it continue to the next loop
+        return true;
+    }
+
     //Now assign the vectorized registers
     assignVectorizedRegisters (info);
 
-- 
1.7.4.1

